### PR TITLE
Add dev.stewlab.PortaWar

### PR DESCRIPTION
**Pull request by stewlab**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: dev.stewlab.PortaWar.desktop [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Version=1.6.0
```
Comment (hfiguiere) [inline dev.stewlab.PortaWar.desktop:2]: this vesion field is invalid. This is NOT the application version.

File: dev.stewlab.PortaWar.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    "app-id": "dev.stewlab.PortaWar",
```
Comment (hfiguiere) [inline dev.stewlab.PortaWar.json:2]: `stewlab.dev` doesn't exist. This app-id doesn't meet the requirements.

File: dev.stewlab.PortaWar.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                {
+                    "type": "file",
+                    "url": "https://github.com/stewlab/flathub/releases/download/v1.6.0/porta-war.pck",
+                    "sha256": "414ad539487f3a2ca3ecd0395438e12c02d72e98b332aaf261d03b2448328d24",
+                    "dest-filename": "porta-war.pck"
+                }
+            ]
+        }
+    ],
+    "finish-args": [
```
Comment (hfiguiere) [inline dev.stewlab.PortaWar.json:26]: finish-args should be above `modules`.

File: .github/pull_request_template.md
Comment (hfiguiere) [inline .github/pull_request_template.md:1]: this should not be here

nor the whole `.github`

**Review by hfiguiere — CHANGES_REQUESTED** [review]
neither the icon, ,desktop nor the appstream files are installed.

File: dev.stewlab.PortaWar.desktop (hunk 1 above)
Comment (stewlab) [reply dev.stewlab.PortaWar.desktop]: resolved

File: dev.stewlab.PortaWar.json (hunk 2 above)
Comment (stewlab) [reply dev.stewlab.PortaWar.json:2]: resolved

File: dev.stewlab.PortaWar.json (hunk 3 above)
Comment (stewlab) [reply dev.stewlab.PortaWar.json]: resolved

File: .github/pull_request_template.md
Comment (stewlab) [reply .github/pull_request_template.md:1]: resolved

**Comment by stewlab** [comment]
All issues should be resolved, thanks!

File: COPYING
Comment (hfiguiere) [inline COPYING:1]: this file doesn't belong here.

File: CONTRIBUTING.md
Comment (hfiguiere) [inline CONTRIBUTING.md:1]: this file doesnt belong here. it's part of the flathub sumbission template

File: dev.stewlab.PortaWar.desktop [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Type=Application
+Name=Porta War
+Comment=Welcome to the war of the handhelds!
+Exec=porta-war
```
Comment (hfiguiere) [inline dev.stewlab.PortaWar.desktop:5]: this is no such executable.

File: dev.stewlab.PortaWar.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{
+    "app-id": "dev.stewlab.PortaWar",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "22.08",
+    "base": "org.godotengine.godot.BaseApp",
+    "base-version": "3.5",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "godot-runner",
```
Comment (hfiguiere) [inline dev.stewlab.PortaWar.json:8]: I don't see any such executable installed either.

Comment (hfiguiere) [reply dev.stewlab.PortaWar.json:8]: and how will it know what to run?

Comment (stewlab) [reply dev.stewlab.PortaWar.json:8]: Ah, I see what you mean.  Fixing now

**Comment by stewlab** [comment]
> neither the icon, ,desktop nor the appstream files are installed.

These should now be installed.  I tested locally, and the desktop files and icons seem to work as intended. They meta info also displays properly in Gnome Software now, thanks.

**Comment by hfiguiere** [comment]
bot, build dev.stewlab.PortaWar

**Comment by hfiguiere** [comment]
appstream file fail to validate:

builddir/files/share/appdata/dev.stewlab.PortaWar.appdata.xml: FAILED:
• tag-missing           : <release> required
• tag-missing           : <url> is not present
Validation of files failed

**Comment by stewlab** [comment]
> Build [50438 failed](https://buildbot.flathub.org/#/builders/6/builds/50438)



> 



> 

aded

> appstream file fail to validate:
> 
> builddir/files/share/appdata/dev.stewlab.PortaWar.appdata.xml: FAILED: • tag-missing : required • tag-missing : is not present Validation of files failed

I just pushed the update with the missing tags, so feel free to rebuild. Thanks.

**Comment by hfiguiere** [comment]
you can do it too

**Comment by stewlab** [comment]
bot, build dev.stewlab.PortaWar

**Comment by stewlab** [comment]
bot, build dev.stewlab.PortaWar

File: dev.stewlab.PortaWar.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    "app-id": "dev.stewlab.PortaWar",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "22.08",
+    "base": "org.godotengine.godot.BaseApp",
+    "base-version": "3.5",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "porta-war",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
```
Comment (hfiguiere) [inline dev.stewlab.PortaWar.json:11]:
```suggestion
        "--socket=x11",
```

**Comment by stewlab** [comment]
bot, build dev.stewlab.PortaWar

File: README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# Flathub
```
[the remark this answers is not in the document]
Comment (stewlab) [reply README.md]: removed, thanks

**Comment by stewlab** [comment]
bot, build dev.stewlab.PortaWar

**Pull request closed without merging**